spark/src/main/kotlin/com/adevinta/spark/res/SparkStringAnnotations.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2023 Adevinta
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.adevinta.spark.res
+
+import android.util.Log
+import androidx.compose.ui.text.SpanStyle
+import com.adevinta.spark.tokens.SparkColors
+import com.adevinta.spark.tokens.SparkTypography
+
+/**
+ * Represents a set of annotations supported by spark that can be used in a string resource.
+ */
+public interface SparkStringAnnotations<T : Any> {
+
+    public fun toSpanStyle(value: String, token: T): SpanStyle?
+
```

Comment:
I'm still not convinced this brings any benefit.
We never use the type SparkStringAnnotations, and subclasses don't hold any value.
IMO this would be way simpler.

```
        public fun Annotation.toSpanStyle(
            key: String,
            value: String,
            colors: SparkColors,
            typography: SparkTypography,
        ): SpanStyle {
            return when (key) {
                "color" -> value.toColorSpanStyle(colors)
                "typography" -> value.toTypographySpanStyle(typography)
                else -> null.also { _ ->
                    Log.d("StringResources", "Annotation $this is not supported by spark")
                }
            }
        }
```